src/Microsoft.Azure.SignalR.Management/Serialization/JsonObjectSerializerHubProtocol.cs

```diff
@@ -0,0 +1,308 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+#if NET5_0_OR_GREATER
+using System.Text.Json.Serialization;
+#endif
+using Azure.Core.Serialization;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.AspNetCore.SignalR.Protocol;
+
+#nullable enable
+
+namespace Microsoft.Azure.SignalR.Management
+{
+    /// <summary>
+    /// Implements the SignalR Hub Protocol using <see cref="ObjectSerializer"/>.
+    /// Modified from https://github.com/dotnet/aspnetcore/blob/d9660d157627af710b71c636fa8cb139616cadba/src/SignalR/common/Protocols.Json/src/Protocol/JsonHubProtocol.cs
```

Comment:
what is modified? can we also comment on the modified part?

Reply:
I have changed the original version so that the diff view on GitHub is more clear. Also added more comments.